mod_aws.c

```diff
@@ -1369,6 +1373,20 @@ static void aws_mod_unload_ev(const void *event_data, void *user_data) {
 }
 #endif /* PR_SHARED_MODULE */
 
+static void aws_postparse_ev(const void *event_data, void *user_data) {
+
+  /* If we should be listening for health checks, AND we have already been
+   * given an explicit AWSHealthCheck address, then start our listener here.
+   * Otherwise, we will wait until we have the EC2 instance' public IPv4
+   * address.
+   */
+  if (aws_use_health == TRUE &&
+      aws_health_addr != NULL) {
+    (void) aws_health_listening(aws_pool, aws_health_addr, aws_health_port,
+      aws_health_uri);
+  }
```

Comment:
By moving this code to a `core.postparse` event handler, we ensure that our `AWSHealthcheck` listener is created even after a SIGHUP.